src/routeDefinitions/useLocalizedRoutes.tsx

```diff
@@ -0,0 +1,297 @@
+import { LangCode, supportedLanguageCodes } from '@/i18n/config';
+import { AuthGuard } from '@/routes/AuthGuard';
+import {
+  AccessibilityStatement,
+  BasicInformation,
+  CookiePolicy,
+  DataSources,
+  PrivacyPolicy,
+  TermsOfService,
+} from '@/routes/BasicInformation';
+import { Home } from '@/routes/Home';
+import {
+  JobOpportunity,
+  Competences as JobOpportunityCompetences,
+  Overview,
+  loader as jobOpportunityLoader,
+} from '@/routes/JobOpportunity';
+import { Favorites, Preferences, Profile, SomethingElse } from '@/routes/Profile';
+import { Competences, competencesLoader } from '@/routes/Profile/Competences';
+import { EducationHistoryV2, loader as educationHistoryLoader } from '@/routes/Profile/EducationHistory';
+import { FreeTimeActivities, loader as freeTimeActivitiesLoader } from '@/routes/Profile/FreeTimeActivities';
+import { WorkHistory, loader as workHistoryLoader } from '@/routes/Profile/WorkHistory';
+import { ErrorElement, Root, loader as rootLoader } from '@/routes/Root';
+import { Goals, Instructions, Interests, Restrictions, Tool, Competences as ToolCompetences } from '@/routes/Tool';
+import {
+  HowDoIGiveFeedback,
+  HowDoIUseTheService,
+  UserGuide,
+  WhatIsTheService,
+  WhereCanIGetMoreHelp,
+  WhoIsTheServiceFor,
+  WhoProvidesTheService,
+} from '@/routes/UserGuide';
+import { useTranslation } from 'react-i18next';
+import { RouteObject, generatePath, matchRoutes, redirect } from 'react-router-dom';
+
+const useLocalizedRoutes = () => {
+  const { i18n } = useTranslation();
+  const jobOpportunityRouteId = (lang: LangCode) => `job-opportunity-${lang}`;
```

Comment:
Maybe this is not the right place for this. It could be another hook or some better mechanism.

Reply:
Created a general useRouteId hook that generates these strings.